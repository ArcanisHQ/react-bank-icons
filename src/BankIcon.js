import React from 'react';
import { propTypes, defaultProps } from './types';
import { getIconByName, getIconById } from './icon-set';

const BankIcon = ({bankName, bankId, size, color}) => {

    const bankIconData = getIconByName(bankName); 
    const svgPath = (color ? bankIconData.svg.mono : bankIconData.svg.original );

    return (
        <svg width={size} className={'rbbi-' + bankIconData.bankName } height={size} viewBox="0 0 24 24" fill={ color } >
            <title>{bankIconData.ariaTitle}</title>
            <desc>{ '√çcone representando a marca do ' + bankIconData.ariaTitle +'.' }</desc>
            <g dangerouslySetInnerHTML={{__html: svgPath }} />
        </svg>
    )}

BankIcon.propTypes = propTypes;
BankIcon.defaultProps = defaultProps;

export default BankIcon;