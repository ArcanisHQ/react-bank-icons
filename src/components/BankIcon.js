import React from "react";
import { getIconById, getIconByName } from "../util/icon-set";
import FallbackIcon from "./FallbackIcon";
import { defaultProps, propTypes } from "./types/BankIcon.types";

const getIcon = (bankName, bankId) => {
  if (bankName && bankId) {
    return getIconByName(bankName);
  }
  return bankId ? getIconById(bankId) : getIconByName(bankName);
};

const BankIcon = ({
  bankName,
  bankId,
  size,
  viewBox = "0 0 32 32",
  color,
  borderRadius,
  fallbackIcon,
}) => {
  console.log({bankId, bankName});
  const bankIconData = getIcon(bankName, bankId);

  const svgBorderRadius = (size * borderRadius) / 24;

  if (!bankIconData) {
    if (fallbackIcon) {
      return (
        <fallbackIcon size={size} borderRadius={borderRadius} color={color} />
      );
    }
    return (
      <FallbackIcon size={size} borderRadius={borderRadius} color={color} />
    );
  }

  const Svg = bankIconData.svg.original;
  const ariaLabel = `√çcone representando a marca do ${bankIconData.ariaTitle}.`;
  console.log(Svg);
  if (typeof Svg === "function") {
    return (
      <svg
        width={size}
        style={{ borderRadius: svgBorderRadius }}
        className={`rbbi-${bankIconData.bankName}`}
        height={size}
        fill={color}
      >
        <title>{bankIconData.ariaTitle}</title>
        <desc>{ariaLabel}</desc>
        <Svg size={size} svgBorderRadius={svgBorderRadius} />
      </svg>
    );
  }

  return (
    <svg
      width={size}
      style={{ borderRadius: svgBorderRadius }}
      className={`rbbi-${bankIconData.bankName}`}
      height={size}
      viewBox="0 0 24 24"
      fill={color}
    >
      <title>{bankIconData.ariaTitle}</title>
      <desc>{ariaLabel}</desc>
      {Svg}
    </svg>
  );
};

BankIcon.propTypes = propTypes;
BankIcon.defaultProps = defaultProps;

export default BankIcon;
