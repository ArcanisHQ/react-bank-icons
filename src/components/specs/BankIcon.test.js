jest.mock('../../icons');
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import BankIcon from '../BankIcon';

const defaultProps = {
  bankName: 'cajuina',
  size: 16,
};

const BankIconComponent = (props = defaultProps) => shallow(<BankIcon {...props} />);
const findChild = (selector) => BankIconComponent().find(selector);

describe('<BankIcon />', () => {
  it('renders BankIcon component correctly', () => {
    const component = renderer.create(<BankIcon {...defaultProps} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('showld show console alert when both BankID and BankName are provided', () => {
    const originalWarn = console.warn;
    const consoleSpy = jest.fn(); 
    console.warn = consoleSpy;
    renderer.create(<BankIcon {...defaultProps} bankId={-11} />);
    expect(consoleSpy).toHaveBeenCalledWith('Both BankID and BankName were provided. BankName will be used.');
    console.warn = originalWarn;
  });  

  it('should do something', () => {
    expect(true).toBeTruthy();
  });
});
