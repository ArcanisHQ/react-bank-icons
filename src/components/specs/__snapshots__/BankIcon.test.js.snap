// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BankIcon /> renders BankIcon component correctly when BankId is provided 1`] = `
<svg
  className="rbbi-cajuina"
  height={16}
  style={
    Object {
      "borderRadius": 1.3333333333333333,
    }
  }
  viewBox="0 0 24 24"
  width={16}
>
  <title>
    Banco Cajuína
  </title>
  <desc>
    Ícone representando a marca do Banco Cajuína.
  </desc>
  <div>
    Cajuína Original.
  </div>
</svg>
`;

exports[`<BankIcon /> renders BankIcon component correctly when BankName is provided 1`] = `
<svg
  className="rbbi-cajuina"
  height={16}
  style={
    Object {
      "borderRadius": 1.3333333333333333,
    }
  }
  viewBox="0 0 24 24"
  width={16}
>
  <title>
    Banco Cajuína
  </title>
  <desc>
    Ícone representando a marca do Banco Cajuína.
  </desc>
  <div>
    Cajuína Original.
  </div>
</svg>
`;

exports[`<BankIcon /> should render FallbackIcon when dont match any icon 1`] = `
<svg
  fill="#3F3F40"
  height={16}
  style={
    Object {
      "borderRadius": 1.3333333333333333,
    }
  }
  viewBox="0 0 24 24"
  width={16}
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    Banco
  </title>
  <desc>
    ícone representando um banco.
  </desc>
  <path
    clipRule="evenodd"
    d="M24 0H0V24H24V0ZM20.7115 9.17332C21.2572 9.57118 20.9767 10.4343 20.305 10.4343H18.957V18.5654H20.5224C20.618 18.5654 20.6963 18.6436 20.6963 18.7393V19.8698C20.6963 19.9655 20.618 20.0438 20.5224 20.0438H3.47753C3.38187 20.0438 3.3036 19.9655 3.3036 19.8698V18.7393C3.3036 18.6436 3.38187 18.5654 3.47753 18.5654H5.04287V10.4343H3.69711C3.02314 10.4343 2.74269 9.569 3.28838 9.17332L11.5912 3.13371C11.7098 3.04683 11.853 3 11.9999 3C12.1469 3 12.2901 3.04683 12.4087 3.13371L20.7115 9.17332ZM6.60821 18.5654H9.15189V10.4343H6.60821V18.5654ZM10.7172 18.5654H13.2609V10.4343H10.7172V18.5654ZM14.8263 18.5654H17.3917V10.4343H14.8263V18.5654Z"
    fillRule="evenodd"
  />
</svg>
`;
